apps/gitness/src/pages-v2/repo/repo-execution-list.tsx

```diff
@@ -0,0 +1,141 @@
+
+import { noop } from 'lodash-es'
+import { parseAsInteger, useQueryState } from 'nuqs'
+
+import { ListExecutionsOkResponse, TypesExecution, useListExecutionsQuery } from '@harnessio/code-service-client'
+import { Icon } from '@harnessio/ui/components'
+import { ExecutionListPage, IExecution, PipelineExecutionStatus } from '@harnessio/ui/views'
+import { timeDistance } from '@harnessio/views'
+
+import Breadcrumbs from '../../components/breadcrumbs/breadcrumbs'
+import { LinkComponent } from '../../components/LinkComponent'
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { useTranslationStore } from '../../i18n/stores/i18n-store'
+import { PathParams } from '../../RouteDefinitions'
+import { PageResponseHeader } from '../../types'
+import { getExecutionStatus } from '../../utils/execution-utils'
+import { useExecutionListStore } from './stores/repo-execution-list-store'
+
+export default function RepoExecutionListPage() {
+  const repoRef = useGetRepoRef()
+  const { pipelineId } = useParams<PathParams>()
+
+  const [query, setQuery] = useQueryState('query')
+  const [queryPage, setQueryPage] = useQueryState('page', parseAsInteger.withDefault(1))
+
+  const { setExecutionsData, page, setPage } = useExecutionListStore()
+
+  const {
+    data: { body: executionsBody, headers } = {},
+    isFetching,
+    isError,
+    error
+  } = useListExecutionsQuery(
+    {
+      repo_ref: repoRef,
+      pipeline_identifier: pipelineId || '',
+      queryParams: { page }
+    },
+    { enabled: !!repoRef }
+  )
+
+  useEffect(() => {
+    if (executionsBody) {
+      const executions = apiExecutions2Executions(executionsBody)
+      const totalPages = parseInt(headers?.get(PageResponseHeader.xTotalPages) || '')
+      setExecutionsData(executions, totalPages)
+    } else {
+      setExecutionsData(null, 0)
+    }
+  }, [executionsBody, headers])
+
+  useEffect(() => {
+    setQueryPage(page)
+  }, [queryPage, page, setPage])
+
+  return (
+    <>
+      <div className="breadcrumbs">
+        <Breadcrumbs />
+      </div>
+      <ExecutionListPage
+        useExecutionListStore={useExecutionListStore}
+        useTranslationStore={useTranslationStore}
+        isLoading={isFetching}
+        isError={isError}
+        errorMessage={error?.message}
+        searchQuery={query}
+        setSearchQuery={setQuery}
+        handleExecutePipeline={noop}
+        LinkComponent={LinkComponent}
+      />
+      {/* TODO */}
+      {/* <RunPipelineDialog
+        open={openRunPipeline}
+        onClose={() => {
+          setOpenRunPipeline(false)
+        }}
+        pipelineId={pipelineId}
+        branch={executions && executions.length > 0 ? executions[0].source : undefined} // TODO: check this
+        toExecutions={'./executions'}
+      /> */}
+    </>
+  )
+}
+
+// NOTE: consider move this function to another file/location
+function apiExecutions2Executions(data: ListExecutionsOkResponse): IExecution[] {
```

Comment:
not blocking, I have created a transform-utils folder, we can keep all of this stuff there for related pages